### PR TITLE
[Issue #11] Make deploy script configurable

### PR DESCRIPTION
**Pull request by pheuberger**
we want to have the deployment configurable. this PR adds support for either a mnemonic via environment variable or a `mnemonic.txt`. the regen bridge and toucan registry addresses can be pulled in via the respective environment vars.

creating this as a draft and when I get sign-off I'll create the documentation for it also in this PR.

### CODE REVIEW

**Comment by pheuberger** [comment]
@kargakis any idea why slither fails?

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export MNEMONIC="flappy bird couch ... mirror"
```
Comment (kargakis) [inline .env.example:1]: No need to use `export`

File: scripts/toucan-bridge.js [hunk 2]
```diff
@@ -1,21 +1,52 @@
...
 
+async function getOwner() {
+	const mnemonic = getMnemonic();
+
+	if (mnemonic) {
+		console.log("Using mnemonic as signer");
+		return ethers.Wallet.fromMnemonic(mnemonic);
+	}
+
+	console.log("Defaulting to first account from `hre.ethers.getSigners()`");
```
Comment (kargakis) [inline scripts/toucan-bridge.js:31]: a bit cleaner
```suggestion
	console.log("Defaulting to first account from Hardhat");
```

**Comment by kargakis** [comment]
> @kargakis any idea why slither fails?

Yes, https://github.com/crytic/slither-action/issues/17

**Comment by kargakis** [comment]
Hrm, don't you need to use `dotenv`?

**Comment by kargakis** [comment]
`dotenv` takes care of loading `.env` into `process.env` for you. We use it in tokenizer as an example. How do you load the env variables now? Also, it would be nice to add a stanza in README explaining how the deploy script works (think the existing documentation will break after your changes)

**Comment by kargakis** [comment]
https://github.com/ToucanProtocol/tokenizer/blob/fe508e6ebf60e03342992982631a345eb42d62a5/packages/hardhat/hardhat.config.ts#L53-L54

**Comment by kargakis** [comment]
> ah right, that's what you mean. yeah, that's a bit down to taste. I have a `dotenv` zsh plugin that loads them when I cd into a folder with a `.env` file. can definitely also implement the `dotenv` npm module +1

Personally in favor of the npm module as it does not impose any additional requirements on the developer wrt tooling.

**Comment by pheuberger** [comment]
> > ah right, that's what you mean. yeah, that's a bit down to taste. I have a `dotenv` zsh plugin that loads them when I cd into a folder with a `.env` file. can definitely also implement the `dotenv` npm module +1
> 
> Personally in favor of the npm module as it does not impose any additional requirements on the developer wrt tooling.

I agree :)

**Comment by pheuberger** [comment]
please have another look @kargakis. I specifically renamed the environment variables to better reflect what they're about as well

**Review by kargakis — APPROVED** [review]

**Pull request merged**